src/qt/askpassphrasedialog.cpp

```diff
@@ -117,53 +122,104 @@ void AskPassphraseDialog::accept()
-                 QMessageBox::Yes|QMessageBox::Cancel,
-                 QMessageBox::Cancel);
-        if(retval == QMessageBox::Yes)
-        {
-            if(newpass1 == newpass2)
+            if(newpass1.empty() || newpass2.empty())
+            {
+                // Cannot encrypt with empty passphrase
+                break;
+            }
+            QMessageBox::StandardButton retval = QMessageBox::question(this, tr("Confirm wallet encryption"),
+                    tr("Warning: If you encrypt your wallet and lose your passphrase, you will <b>LOSE ALL OF YOUR NAVCOINS</b>!") + "<br><br>" + tr("Are you sure you wish to encrypt your wallet?"),
+                    QMessageBox::Yes|QMessageBox::Cancel,
+                    QMessageBox::Cancel);
+            if(retval == QMessageBox::Yes)
             {
-                if(model->setWalletEncrypted(true, newpass1))
+                if(newpass1 == newpass2)
                 {
-                    QMessageBox::warning(this, tr("Wallet encrypted"),
-                                         "<qt>" +
-                                         tr("%1 will close now to finish the encryption process. "
-                                         "Remember that encrypting your wallet cannot fully protect "
-                                         "your navcoins from being stolen by malware infecting your computer.").arg(tr(PACKAGE_NAME)) +
-                                         "<br><br><b>" +
-                                         tr("IMPORTANT: Any previous backups you have made of your wallet file "
-                                         "should be replaced with the newly generated, encrypted wallet file. "
-                                         "For security reasons, previous backups of the unencrypted wallet file "
-                                         "will become useless as soon as you start using the new, encrypted wallet.") +
-                                         "</b></qt>");
-                    QApplication::quit();
+                    if(model->setWalletEncrypted(true, newpass1))
+                    {
+                        QMessageBox::warning(this, tr("Wallet encrypted"),
+                                "<qt>" +
+                                tr("%1 will close now to finish the encryption process. "
+                                    "Remember that encrypting your wallet cannot fully protect "
+                                    "your navcoins from being stolen by malware infecting your computer.").arg(tr(PACKAGE_NAME)) +
+                                "<br><br><b>" +
+                                tr("IMPORTANT: Any previous backups you have made of your wallet file "
+                                    "should be replaced with the newly generated, encrypted wallet file. "
+                                    "For security reasons, previous backups of the unencrypted wallet file "
```

Comment:
This is no longer applicable "will become useless as soon as you start using the new, encrypted wallet"

Since we disabled that feature to regenerate the master